packages/demand-bloc-payment/src/providers/AdyenProvider.test.ts

```diff
@@ -308,25 +305,28 @@ describe('AdyenProvider', () => {
       expect(cardElement.handleAction).toHaveBeenCalledTimes(1)
     })
 
-    it('should do nothing if tokenizeHostedFields have not been called', () => {
-      provider.startThreeDSecureVerification()
-
-      expect(cardElement.handleAction).toHaveBeenCalledTimes(0)
+    it('should do nothing if tokenizeHostedFields have not been called', async () => {
+      try {
+        await provider.startThreeDSecureVerification()
+      } catch (e) {
+        expect(cardElement.handleAction).toHaveBeenCalledTimes(0)
+      }
     })
 
-    it('should do nothing if handleAction is not a function', () => {
-      provider.startThreeDSecureVerification()
-
-      expect(cardElement.handleAction).toHaveBeenCalledTimes(0)
+    it('should do nothing if handleAction is not a function', async () => {
+      try {
+        await provider.startThreeDSecureVerification()
+      } catch (e) {
+        expect(cardElement.handleAction).toHaveBeenCalledTimes(0)
+      }
     })
 
     it('should emit error if no action is defined', async () => {
-      await provider.tokenizeHostedFields()
-      provider.paymentAction = null
-
       try {
-        await provider.startThreeDSecureVerification()
+        await provider.tokenizeHostedFields()
+        provider.paymentAction = null
       } catch (error) {
+        await provider.startThreeDSecureVerification()
```

Comment:
why this is in catch?
is test setup correct - `tokenizeHostedFields` throws error and `startThreeDSecureVerification` tested in catch block. Looks strange